enterprise/server/backends/migration_cache/migration_cache.go

```diff
@@ -736,8 +734,40 @@ func (mc *MigrationCache) Start() error {
 func (mc *MigrationCache) Stop() error {
 	log.Info("Migration cache beginning shut down")
 	close(mc.quitChan)
+	// Wait for migration-related channels that use cache resources (like copying) to close before shutting down
+	// the caches
 	if err := mc.eg.Wait(); err != nil {
 		return err
 	}
-	return nil
+
+	var wg sync.WaitGroup
+	var srcShutdownErr, dstShutdownErr error
+	if src, canStopSrc := mc.src.(interfaces.StoppableCache); canStopSrc {
```

Comment:
@vadimberezniker  Just to make sure I interpreted the docs correctly - when I cast mc.src here and call src.Stop(), it's calling Stop on mc.src right? It's not copying mc.src -> src and only calling Stop on the copy? 

Reading stuff like this : 
https://stackoverflow.com/questions/13511203/why-cant-i-assign-a-struct-to-an-interface

Reply:
That's correct. You're effectively calling methods on the original mc.src

Reply:
Great - thanks for confirming